website/docs/language/functions/uuid.html.md

```diff
@@ -11,15 +11,17 @@ description: |-
 `uuid` generates a unique identifier string.
 
 The id is a generated and formatted as required by
-[RFC 4122 section 4.4](https://tools.ietf.org/html/rfc4122#section-4.4),
+[RFC 4122 section 4.4](https://datatracker.ietf.org/doc/html/rfc4122#section-4.4),
 producing a Version 4 UUID. The result is a UUID generated only from
 pseudo-random numbers.
 
 This function produces a new value each time it is called, and so using it
 directly in resource arguments will result in spurious diffs. We do not
 recommend using the `uuid` function in resource configurations, but it can
 be used with care in conjunction with
-[the `ignore_changes` lifecycle meta-argument](/docs/language/meta-arguments/lifecycle.html#ignore_changes).
+[the `ignore_changes` lifecycle meta-argument](/docs/language/meta-arguments/lifecycle.html#ignore_changes). 
+
+If this function produced a value during the plan step, it would cause the final configuration during the apply step not to match the actions shown in the plan (since the function is called again in the apply step, and would return a different value), which violates the Terraform execution model. For that reason, this function produces an unknown value result during the plan step, with the real result being decided only during the apply step.
```

Comment:
```suggestion
Terraform cannot predict the result of `uuid` during the planning phase, so the function produces an unknown value in the plan. This prevents Terraform from creating a final configuration that does not match the plan.

```

Reply:
Thanks for the suggestions! Obviously this is your documentation, so you can edit it however you see fit. IMHO though, I think it would benefit from clarity on _why_ Terraform can't (well, _shouldn't_, because it's a conscious choice not to, not a technical limitation) determine the value of `uuid()` during the plan step. I think it would be helpful to explicitly describe that the function gets run twice during an apply (once with an `unknown` result during the plan step, and again with the real value during the apply step).